src/Symfony/Component/Serializer/Serializer.php

```diff
@@ -202,8 +205,21 @@ public function supportsDenormalization($data, $type, $format = null, array $con
      */
     private function getNormalizer($data, $format, array $context)
     {
+        if (\is_object($data) && isset($this->normalizerCache[\get_class($data)][$format])) {
+            return $this->normalizerCache[\get_class($data)][$format];
+        }
+
+        $cacheable = true;
         foreach ($this->normalizers as $normalizer) {
+            if ($cacheable) {
```

Comment:
think this needs to be inside the `$normalizer instanceof NormalizerInterface` check

Reply:
No, if a normalizer in the chain before the matching one doesn't implement this new interface, we cannot use the cache (or it will be a BC break).

Reply:
but we are only interested in `NormalizerInterface` normalizers here? some only implement `DenormalizerInterface` like `ArrayDenormalizer`?

Reply:
Ok got it, sorry.